Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -150,13 +152,19 @@
     
 	% If tokenVisitAliases is not provided, we assume visits are not renamed and create a dummy tokenVisitAliases list
 	if ~isfield(x.modules.import.imPar, 'tokenVisitAliases') || isempty(x.modules.import.imPar.tokenVisitAliases)
-		% Create a unique list of visits
-		listVisits = unique(listVisits);
-	
-		% Create a list of unique visits names
-		if ~isempty(listVisits)
-			x.modules.import.imPar.tokenVisitAliases = listVisits(:);
-			x.modules.import.imPar.tokenVisitAliases(:,2) = x.modules.import.imPar.tokenVisitAliases(:,1);
+		if bVisitsDetected
+			% Create a unique list of visits
+			listVisits = unique(listVisits);
+
+			% Create a list of unique visits names
+			if ~isempty(listVisits)
+				x.modules.import.imPar.tokenVisitAliases = listVisits(:);
+				x.modules.import.imPar.tokenVisitAliases(:,2) = x.modules.import.imPar.tokenVisitAliases(:,1);
+			end
+		else
```

Comment:
I would in general suggest that we start with the default/fallback case for an if-statement (which is usually a small part -> we default/fallback to empty something) but then the true case can give a page of code. So easier to read the default/fallback version first and then read what happens in the true case (long code).

Reply:
OK. No prob. I have switched the statements. 

Done. You can resolve.